cimi-resources/test/com/sixsq/slipstream/ssclj/resources/external_object_public_lifecycle_test.clj

```diff
@@ -0,0 +1,133 @@
+
+
+(deftest lifecycle
+  (eoltu/full-eo-lifecycle (str p/service-context eot/resource-url "/" public/objectType)
+                           (external-object)))
+
+
+(defn throw-object-acl-does-not-exist [_ _ _]
+  (let [ex (doto
+             (AmazonServiceException. "Simulated AWS Exception for missing public read ACL")
+             (.setStatusCode 404))]
+    (throw ex)))
+
+
+(deftest check-public-access
+  (let [session-anon (-> (ltu/ring-app)
+                         session
+                         (content-type "application/json"))
+        session-user (header session-anon authn-info-header eoltu/user-info-header)
+        base-uri (str p/service-context (u/de-camelcase eo/resource-name))
+        template (eoltu/get-template (str p/service-context eot/resource-url "/" public/objectType))
+        create-href {:externalObjectTemplate (-> (external-object)
+                                                 (assoc :href (:id template))
+                                                 (dissoc :objectType))}]
+
+    ;; Create the test object.
+    (-> session-user
+        (request base-uri
+                 :request-method :post
+                 :body (json/write-str create-href))
+        (ltu/body->edn)
+        (ltu/is-status 201)
+        (ltu/location))
+
+    (let [entry (-> session-user
+                    (request base-uri)
+                    (ltu/body->edn)
+                    (ltu/is-status 200)
+                    (ltu/is-resource-uri eo/collection-uri)
+                    (ltu/is-count 1)
+                    (ltu/entries eo/resource-tag)
+                    first)
+          id (:id entry)
+          abs-uri (str p/service-context id)
+          upload-op (-> session-user
+                        (request abs-uri)
+                        (ltu/body->edn)
+                        (ltu/is-operation-present "upload")
+                        (ltu/is-operation-present "delete")
+                        (ltu/is-operation-present "edit")
+                        (ltu/is-operation-absent "ready")
+                        (ltu/is-operation-absent "download")
+                        (ltu/is-status 200)
+                        (ltu/get-op "upload"))
+
+          abs-upload-uri (str p/service-context (u/de-camelcase upload-op))]
+
+      ;; Upload object's content.
+      (-> session-user
+          (request abs-upload-uri
+                   :request-method :post)
+          (ltu/body->edn)
+          (ltu/is-status 200))
+
+      (let [uploading-eo (-> session-user
+                             (request abs-uri)
+                             (ltu/body->edn)
+                             (ltu/is-operation-present "ready")
+                             (ltu/is-status 200))
+
+            ready-url-action (str p/service-context (ltu/get-op uploading-eo "ready"))]
+
+
+        ;; Missing ACL should fail the action
+        (with-redefs [s3/set-acl-public-read throw-object-acl-does-not-exist]
```

Comment:
Is this really the expected behavior from the AWS S3 client?  I would expect that the method would always return an ACL, whether the object is public or not.